api/actions/actionsMiddleware.js

```diff
@@ -0,0 +1,12 @@
+const validateSubjectBody = (req, res, next) => {
```

Comment:
My only suggestion would be that we add some kind of check(s) that makes sure no one is trying to do anything malicious. For example, a text limit so someone can just copy-paste a massive block of text.

Reply:
Kade and I added some valiation constraints.